### PR TITLE
Add AdvisorTestRunner and replace TestController

### PR DESCRIPTION
**Pull request by ziyuezh576**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                })
+                .WithNewRecordMatcherArguments (
+                    userAgentsToIgnore: new Dictionary<string, string>(),
+                    resourceProviders: new Dictionary<string, string>
+                    {
+                        {"Microsoft.Resources", null},
+                        {"Microsoft.Features", null},
+                        {"Microsoft.Authorization", null}
+                    }
+                ).WithManagementClients(
```
Comment (vidai-msft) [inline src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs:36]: WithManagementClients is not necessary as both GetResourceGraphClient and GetSubscriptionClient instantiate the object like context.GetServiceClient<TClient>

File: src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    GetResourceGraphClient,
+                    GetSubscriptionClient
+                )
+                .Build();
+        }
+
+        private static IAdvisorManagementClient GetResourceGraphClient(MockContext context)
```
Comment (vidai-msft) [inline src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs:43]: This method is not necessary. Please remove it.

File: src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            return context.GetServiceClient<AdvisorManagementClient>(TestEnvironmentFactory.GetTestEnvironment());
+        }
+
+        private static SubscriptionClient GetSubscriptionClient(MockContext context)
```
Comment (vidai-msft) [inline src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs:48]: This method is not necessary. Please remove it

**Review by vidai-msft — CHANGES_REQUESTED** [review]
Please modify the AdvisorTestRunner as per the comments below.

**Review by vidai-msft — APPROVED** [review]
Looks good !

File: src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
```
Comment (isra-fel) [inline src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs:1]: Please add license headers - you can copy it from other files

**Review by isra-fel — CHANGES_REQUESTED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**